### DISCUSSION TITLE
Convert timestamp into human-readable duration in jinja2

### DISCUSSION THREAD

**Discussion by thecatontheflat** [jertel/elastalert2#1381] [Q&A] [answered]
Hi team,

I am trying to send the "outage duration" as part of the alert message, e.g.:

```
alert_text: |
  {% if monitor.status == "up" %}
    :green_apple: <{{ url.full }}|{{ monitor.name }}> is {{ monitor.status|upper }} (duration {{ state.ends.duration_ms | int | datetimeformat('%H:%M') }}) :green_apple:
  {% else %}
    :everythings_fine_parrot: <{{ url.full }}|{{ monitor.name }}> is {{ monitor.status|upper }} :everythings_fine_parrot:
  {% endif %}
```

The problem is that there is no `datetimeformat` filter. Could you please point me in the correct direction regarding converting the timestamp into "duration" (e.g 6h 23m)? 

I assume I should I create an `Enhancement` for this, but the problem seemed trivial enough that I just wanted to check first if there is another (built-in?) way of achieving it.

**Comment by jertel** [comment, marked as answer]
Yes, since Jinja2 doesn't offer a built-in filter for this we have to do the time string manipulation ourselves. It seems every few months someone posts a discussion asking for help converting their timestamp into a particular format or timezone, and we've been recommending enhancements for that, as you suspected. There are likely some examples in the discussion forum.

**Reply by thecatontheflat** [reply]
Got it, thanks!
